core/services/repo_manager.rb

```diff
@@ -123,8 +123,18 @@ def find_repository(product_name, product, box)
     version = product['version'].nil? ? 'default' : product['version']
     repository_key = $session.box_definitions.platform_key(box)
     repository_name = PRODUCT_ATTRIBUTES[product_name][:repository]
-    repo_key = "#{repository_name}@#{version}+#{repository_key}"
-    repo = @repos[repo_key]
+    if product['version'].nil?
```

Comment:
The intent of the code is not clear. Please extract it to the private method and supply the comment on the method intent.